kin-sdk/src/test/java/kin/sdk/TransactionSenderTest.java

```diff
@@ -134,6 +133,28 @@ public void sendTransaction_ToAccountNotExist() throws Exception {
         transactionSender.sendTransaction(transaction);
     }
 
+	@Test
+	public void sendTransaction_Http_307_Response_Success() throws Exception {
+		//send transaction fetch first to account details, then from account details, and finally perform tx which will return 307
+		// and then 200. here we mock all 4 responses from server to achieve success operation
+		mockWebServer.enqueue(TestUtils.generateSuccessMockResponse(this.getClass(), "tx_account_from.json"));
+		mockWebServer.enqueue(TestUtils.generateSuccessMockResponse(this.getClass(), "tx_account_to.json"));
+		// No need for a real location because any way it is local host
+		mockWebServer.enqueue(TestUtils.generateSuccessHttp307MockResponse(this.getClass(), "tx_success_res.json", ""));
+		mockWebServer.enqueue(TestUtils.generateSuccessMockResponse(this.getClass(), "tx_success_res.json"));
+
+		Transaction transaction = transactionSender
+			.buildTransaction(account, ACCOUNT_ID_TO, new BigDecimal("1.5"), FEE);
+		TransactionId transactionId = transactionSender.sendTransaction(transaction);
+
+		assertEquals("8f1e0cd1d922f4c57cc1898ececcf47375e52ec4abf77a7e32d0d9bb4edecb69", transactionId.id());
+
+		//verify sent requests data
+		assertThat(mockWebServer.takeRequest().getRequestUrl().toString(), containsString(ACCOUNT_ID_FROM));
+		assertThat(mockWebServer.takeRequest().getRequestUrl().toString(), containsString(ACCOUNT_ID_TO));
+		assertThat(mockWebServer.takeRequest().getBody().readUtf8(), equalTo(TX_BODY));
```

Comment:
should assert for additional (4th) request that includes the body (after the 307 redirection)

Reply:
fixed